src/amo/components/SearchForm/index.js

```diff
@@ -0,0 +1,241 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import Autosuggest from 'react-autosuggest';
+import { withRouter } from 'react-router';
+import defaultDebounce from 'simple-debounce';
+
+import {
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_THEME,
+  validAddonTypes,
+} from 'core/constants';
+import log from 'core/logger';
+import translate from 'core/i18n/translate';
+import { convertFiltersToQueryParams } from 'core/searchUtils';
+import SearchInput from 'ui/components/SearchInput';
+import { withErrorHandler } from 'core/errorHandler';
+import { getAddonIconUrl } from 'core/imageUtils';
+import {
+  autocompleteCancel,
+  autocompleteStart,
+} from 'core/reducers/autocomplete';
+import Suggestion from 'amo/components/SearchForm/Suggestion';
+
+import 'core/css/inc/lib.scss';
+import './styles.scss';
+
+
+export class SearchFormBase extends React.Component {
+  static propTypes = {
+    addonType: PropTypes.string,
+    api: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+    pathname: PropTypes.string.isRequired,
+    query: PropTypes.string.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    errorHandler: PropTypes.object.isRequired,
+    suggestions: PropTypes.arrayOf(PropTypes.shape({
+      name: PropTypes.string.isRequired,
+      url: PropTypes.string.isRequired,
+      iconUrl: PropTypes.string.isRequired,
+    })).isRequired,
+    loadingSuggestions: PropTypes.bool.isRequired,
+    router: PropTypes.object.isRequired,
+    debounce: PropTypes.func.isRequired,
+  }
+
+  static defaultProps = {
+    debounce: defaultDebounce,
+  };
+
+  constructor(props: Object) {
+    super(props);
+
+    this.state = {
+      searchValue: props.query || '',
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { query } = nextProps;
+
+    if (this.props.query !== query) {
+      this.setState({ searchValue: query || '' });
+    }
+  }
+
+  getSuggestions() {
+    if (this.props.loadingSuggestions) {
+      // 10 is the maximum number of results returned by the API
+      return Array(10).fill({
+        name: 'loading-text',
+        iconUrl: getAddonIconUrl(),
+        loading: true,
+      });
+    }
+
+    return this.props.suggestions.map((suggestion) => ({
+      ...suggestion,
+      loading: false,
```

Comment:
I wanted to keep a consistent 
 shape for the Autosuggest `suggestions`.

Reply:
Also, it does not make really sense to have `loading` in the reducer, because it is a pure UI prop. That is why I think it makes sense to augment the `suggestions` in the methods used by the `Autosuggest` component.